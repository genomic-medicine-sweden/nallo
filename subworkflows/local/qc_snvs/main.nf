//
// Workflow to QC VCFs
//
include { DEEPVARIANT_VCFSTATSREPORT } from '../../../modules/nf-core/deepvariant/vcfstatsreport/main'
include { BCFTOOLS_STATS             } from '../../../modules/nf-core/bcftools/stats/main'

workflow QC_SNVS {
    take:
    ch_vcf              // channel: [mandatory] [Â val(meta), path(vcf) ]
    ch_normalized_vcf   // channel: [mandatory] [ val(meta), path(vcf) ]
    ch_normalized_index // channel: [mandatory] [ val(meta), path(vcf) ]

    main:
    ch_versions = Channel.empty()

    DEEPVARIANT_VCFSTATSREPORT(
        ch_vcf
    )
    ch_versions = ch_versions.mix(DEEPVARIANT_VCFSTATSREPORT.out.versions)

    BCFTOOLS_STATS(
        ch_normalized_vcf.join(
            ch_normalized_index,
            failOnMismatch:true,
            failOnDuplicate:true
        ),
        [[],[]],
        [[],[]],
        [[],[]],
        [[],[]],
        [[],[]]
    )
    ch_versions = ch_versions.mix(BCFTOOLS_STATS.out.versions)

    emit:
    vcfstatsreport = DEEPVARIANT_VCFSTATSREPORT.out.report // channel: [ val(meta), path(html) ]
    stats          = BCFTOOLS_STATS.out.stats              // channel: [ val(meta), path(txt) ]
    versions       = ch_versions                           // channel: [ path(versions.yml) ]
}
