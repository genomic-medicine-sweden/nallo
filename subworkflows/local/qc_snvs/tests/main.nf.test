nextflow_workflow {

    name "Test Workflow QC_SNVS"
    script "../main.nf"
    workflow "QC_SNVS"

    test("DeepVariant VCF, Normalized VCF and index") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id: 'test' ],
                    file(params.pipelines_testdata_base_path + 'testdata/HG002_deepvariant.vcf.gz', checkIfExists: true),
                ])
                input[1] = Channel.of([
                    [ id: 'test' ],
                    file(params.pipelines_testdata_base_path + 'testdata/HG002_deepvariant_norm.vcf.gz', checkIfExists: true),
                ])
                input[2] = Channel.of([
                    [ id: 'test' ],
                    file(params.pipelines_testdata_base_path + 'testdata/HG002_deepvariant_norm.vcf.gz.tbi', checkIfExists: true),
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out,
                ).match() }
            )
        }
    }

    test("DeepVariant VCF, Normalized VCF and index -stub") {

        options "-stub"

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id: 'test' ],
                    file(params.pipelines_testdata_base_path + 'testdata/HG002_deepvariant.vcf.gz', checkIfExists: true),
                ])
                input[1] = Channel.of([
                    [ id: 'test' ],
                    file(params.pipelines_testdata_base_path + 'testdata/HG002_deepvariant_norm.vcf.gz', checkIfExists: true),
                ])
                input[2] = Channel.of([
                    [ id: 'test' ],
                    file(params.pipelines_testdata_base_path + 'testdata/HG002_deepvariant_norm.vcf.gz.tbi', checkIfExists: true),
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out,
                ).match() }
            )
        }
    }
}
