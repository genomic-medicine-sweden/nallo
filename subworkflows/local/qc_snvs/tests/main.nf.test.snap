{
    "DeepVariant VCF, Normalized VCF and index": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.visual_report.html:md5,8821e36a7d1b30efe65751dbacb28170"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcftools_stats.txt:md5,a84bf91b4f074e3c07f23bd285dcc12b"
                    ]
                ],
                "2": [
                    "versions.yml:md5,55c62b669824d245bb9d6b9ff8a139ab",
                    "versions.yml:md5,e81049e4d6445b1aa9bb4c5759c4f72c"
                ],
                "stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcftools_stats.txt:md5,a84bf91b4f074e3c07f23bd285dcc12b"
                    ]
                ],
                "vcfstatsreport": [
                    [
                        {
                            "id": "test"
                        },
                        "test.visual_report.html:md5,8821e36a7d1b30efe65751dbacb28170"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,55c62b669824d245bb9d6b9ff8a139ab",
                    "versions.yml:md5,e81049e4d6445b1aa9bb4c5759c4f72c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-22T11:09:45.43211778"
    },
    "DeepVariant VCF, Normalized VCF and index -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.visual_report.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcftools_stats.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,55c62b669824d245bb9d6b9ff8a139ab",
                    "versions.yml:md5,e81049e4d6445b1aa9bb4c5759c4f72c"
                ],
                "stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bcftools_stats.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vcfstatsreport": [
                    [
                        {
                            "id": "test"
                        },
                        "test.visual_report.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,55c62b669824d245bb9d6b9ff8a139ab",
                    "versions.yml:md5,e81049e4d6445b1aa9bb4c5759c4f72c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-22T11:09:52.29171863"
    }
}